stellarphot/photometry/source_detection.py

```diff
@@ -127,29 +104,56 @@ def compute_fwhm(
         except AttributeError:
             pass
 
-        cutout = Cutout2D(ccd.data, (x, y), 5 * fwhm_estimate)
-        if fit:
-            fit = _fit_2dgaussian(cutout.data)
-            fwhm_x.append(gaussian_sigma_to_fwhm * fit.x_stddev_1)
-            fwhm_y.append(gaussian_sigma_to_fwhm * fit.y_stddev_1)
-            # print('Still fitting!!')
+        # SKY SUBTRACT SHIT!!
+        cutout = Cutout2D(ccd.data - sky, (x, y), 5 * fwhm_estimate)
+
+        # Mask any NaNs in the data
+        nan_mask = np.isnan(cutout.data)
+        inp_mask = getattr(ccd, "mask", False)
+        if inp_mask:
+            inp_mask = inp_mask[cutout.slices_original()]
+            mask = inp_mask | nan_mask
         else:
-            sc = data_properties(cutout.data - sky)
-
-            # dat = np.where(cutout.data - sky > 0, cutout.data - sky, 0)
-            # mom1 = _moments(dat, order=1)
-            # xc = mom1[0, 1] / mom1[0, 0]
-            # yc = mom1[1, 0] / mom1[0, 0]
-            # moments = _moments_central(dat,
-            #                            center=(xc, yc), order=2)
-            # mom_scale = (moments / mom1[0, 0])
-            # fwhm_xm = 2 * np.sqrt(np.log(2) * mom_scale[0, 2])
-            # fwhm_ym = 2 * np.sqrt(np.log(2) * mom_scale[2, 0])
-
-            fwhm_xm = sc.fwhm.value
-            fwhm_ym = fwhm_xm
-            fwhm_x.append(fwhm_xm)
-            fwhm_y.append(fwhm_ym)
+            mask = nan_mask
+
+        cutout_xy = cutout.to_cutout_position((x, y))
+        match fit_method:
+            case FwhmMethods.FIT:
+                # Make sure we get an odd fits shape
+                fit_shape = int(2 * ((5 * fwhm_estimate) // 2) + 1)
+
+                # fit_fwhm is supposed to handle NaNs automatically but it doesn't
+                # as of photutils 2.2.0
+                # see https://github.com/astropy/photutils/issues/2029
+                # For now replace any NaN with zero and hope for the best.
+                cutout.data[nan_mask] = 0
```

Comment:
Given these are masked, isn't there a way to replace with the mean of the surrounding pixels?  Kinda surprised, as that would seem that a safer approach.

I did a quick search of the astropy documentation and found 
```
from astropy.convolution import interpolate_replace_nans
result = interpolate_replace_nans(image, kernel)
```
is supposed to use convolution to replace NaNs, that might be a safer approach.

Reply:
At a minimum it can't hurt to do that unless it winds up being slow. The bug has been fixe in photutils and there it is simply ignored.

Reply:
I'm leaning against this for now because photutils will, once the next version is out, ignore this value anyway.